### PR TITLE
Issue 73 - Refactor building ObsRequest optional filter parameters

### LINKED ISSUE #73: [API] Clean up how we build the GetObsRequest filter when there are optional parameters
In `edr.py`, correctly handle the case of `parameter_name=None`, see lines 165 and 268. Probably best to build the filter dict, only adding something if it is provided. Do the same for other optional parameters.

### PR DESCRIPTION
**Pull request by fjugipe**
Closes #73 

- Build filter & datetime only when needed for obs request

Note /locations request filtering already fixed in #66

### CODE REVIEW

File: api/routers/edr.py [hunk 1]
```diff
@@ -158,18 +158,22 @@ async def get_data_location_id(
...
     )
+
+    if parameter_name:
+        parameter_name = split_and_strip(parameter_name)
+        await verify_parameter_names(parameter_name)
+        request.filter["parameter_name"].values.extend(parameter_name)
+
+    if datetime:
+        range = get_datetime_range(datetime)
+        request.temporal_interval.CopyFrom(dstore.TimeInterval(start=range[0], end=range[1]))
```
Comment (lukas-phaf) [inline api/routers/edr.py:175]: You can't assign messages fields directly, and this is why you use this "trick", right?

Accroding to [this](https://github.com/EURODEO/e-soh/pull/77), you can also set `start` and `end` fields directly, would that be clearer?

Comment (fjugipe) [reply api/routers/edr.py:175]: You can't directly assign a new value. That is why I made a new TimeInterval object and used `CopyFrom` to assign to it at the same time. If the values or fields are already there, then they can be assigned.

I could make it like this, if you think it is more clearer? 
```suggestion
        request.temporal_interval.CopyFrom(dstore.TimeInterval())
        request.temporal_interval.start = range[0]
        request.temporal_interval.end = range[1]
```

Comment (lukas-phaf) [reply api/routers/edr.py:175]: Sorry, put the wrong link *and* was reading it wrong.

The link is:
https://protobuf.dev/reference/python/python-generated/#embedded_message

This works:
```
        request.temporal_interval.start.CopyFrom(range[0])
        request.temporal_interval.end.CopyFrom(range[1])
```

I'll leave it up to you to what you like best. But I would change the first line to tuple unpacking, so something like:
```
        start, end = get_datetime_range(datetime)
        request.temporal_interval.start.CopyFrom(start)
        request.temporal_interval.end.CopyFrom(end)
```

Comment (fjugipe) [reply api/routers/edr.py]: I see your point now and refactored it 👍

**Review by lukas-phaf — APPROVED** [review]

**Pull request merged**